### PR TITLE
P1892R1 Extended locale-specific presentation specifiers for std::format

### LINKED ISSUE #3424: [2019-11 LWG Motion 22] 1892R1 Extended locale-specific presentation specifiers for std::format
http://wiki.edg.com/pub/Wg21belfast/StrawPolls/P1892R1.pdf

This resolves NB comment GB226

### PR DESCRIPTION
**Pull request by burblebee**
Fixes #3424.

Fixes cplusplus/nbballot#223
Fixes cplusplus/papers#641

### CODE REVIEW

File: source/utilities.tex [hunk 1]
```diff
@@ -19886,6 +19886,30 @@
...
+\pnum
+When the \tcode{L} option is used, the form used for the conversion
+is called the \defn{locale-specific} form.
+The \tcode{L} option is only valid for arithmetic types, and
+its effect depends upon the type.
+\begin{itemize}
+
+\item
+For integral types, the locale-specific form
+causes the context’s locale to be used
```
Comment (johelegp) [inline source/utilities.tex]:
```suggestion
causes the context's locale to be used
```

File: source/utilities.tex [hunk 2]
```diff
@@ -19886,6 +19886,30 @@
...
+to insert the appropriate digit group separator characters.
+
+\item
+For floating-point types, the locale-specific form
+causes the context’s locale to be used
```
Comment (johelegp) [inline source/utilities.tex]:
```suggestion
causes the context's locale to be used
```

File: source/utilities.tex [hunk 3]
```diff
@@ -19886,6 +19886,30 @@
...
+to insert the appropriate digit group and decimal radix separator characters.
+
+\item
+For the textual representation of \tcode{bool}, the locale-specific form
+causes the context’s locale to be used
```
Comment (johelegp) [inline source/utilities.tex]:
```suggestion
causes the context's locale to be used
```

File: source/utilities.tex [hunk 4]
```diff
@@ -19886,6 +19886,30 @@
 For floating-point types this field specifies the formatting precision.
 For string types it specifies how many characters will be used from the string.
 
+\pnum
+When the \tcode{L} option is used, the form used for the conversion
+is called the \defn{locale-specific} form.
```
Comment (jensmaurer) [inline source/utilities.tex]: This just puts "local-specific" into the index. I think we want to define "locale-specific form" and use `\defnadj{locale-specific}{form}`

File: source/utilities.tex [hunk 5]
```diff
@@ -19886,6 +19886,30 @@
...
+
+\item
+For integral types, the locale-specific form
+causes the context’s locale to be used
+to insert the appropriate digit group separator characters.
+
+\item
+For floating-point types, the locale-specific form
+causes the context’s locale to be used
+to insert the appropriate digit group and decimal radix separator characters.
```
Comment (jensmaurer) [inline source/utilities.tex]: The examples in the paper show the "decimal" radix separator character to be used even with hexfloats. Something doesn't add up here.

**Comment by jensmaurer** [comment]
... and please revert the lines in the commit comment: Pxxxx first, then "Fixes NB US 123" in the body.

File: source/utilities.tex (hunk 5 above)
Comment (peter-b) [reply source/utilities.tex]: @jensmaurer This was a drafting error on my part and I'm sorry it didn't get picked up.

The examples are correct and the wording is is not correct.  I suggest `s/decimal//`.

The intention is to give access the features currently available when using iostreams, e.g.

```c++
#include <iostream>
#include <locale>

int main() {
    const std::locale gb{"en_GB"};
    std::cout.imbue(gb);
    std::cout << std::hex << 123456 << "\n";
    return 0;
}
```
which produces `1e,240` in implementations I have access to right now.

File: source/utilities.tex (hunk 3 above)
Comment (burblebee) [reply source/utilities.tex]: I guess this a Unicode issue?  can't see any difference on a mac, so thanks for catching these!

File: source/utilities.tex (hunk 4 above)
Comment (burblebee) [reply source/utilities.tex]: yes, much better.  Thanks!

File: source/utilities.tex (hunk 5 above)
Comment (burblebee) [reply source/utilities.tex]: @zygoloid, are you ok with me making this normative change editorially?

Comment (peter-b) [reply source/utilities.tex]: @burblebee @zygoloid If necessary, I will raise an LWG issue.

Comment (zygoloid) [reply source/utilities.tex]: I think the intent is sufficiently clear, especially given the description given in the paper.

**Pull request merged**